apis/database/v1alpha1/redshift_types.go

```diff
@@ -0,0 +1,713 @@
+	StateFailed = "failed"
+)
+
+// +kubebuilder:object:root=true
+
+// An Redshift is a managed resource that represents an AWS Redshift cluster.
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="STATE",type="string",JSONPath=".status.atProvider.clusterStatus"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,managed,aws}
+type Redshift struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RedshiftSpec   `json:"spec"`
+	Status RedshiftStatus `json:"status,omitempty"`
+}
+
+// An RedshiftSpec defines the desired state of an Redshift.
+type RedshiftSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  RedshiftParameters `json:"forProvider"`
+}
+
+// RedshiftParameters define the desired state of an AWS Relational Database
+// Service cluster.
+type RedshiftParameters struct {
+	// AllowVersionUpgrade indicates that major engine upgrades are applied automatically to the DB
+	// cluster during the maintenance window.
+	// default=true
+	// +optional
+	AllowVersionUpgrade *bool `json:"allowVersionUpgrade,omitempty"`
+
+	// AutomatedSnapshotRetentionPeriod is the number of days for which
+	// automated backups are retained. Setting this parameter to a positive
+	// number enables backups. Setting this parameter to  0 disables automated backups.
+	// default=1
+	// +kubebuilder:validation:Maximum=35
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	AutomatedSnapshotRetentionPeriod *int64 `json:"automatedSnapshotRetentionPeriod,omitempty"`
+
+	// AvailabilityZone is the EC2 Availability Zone in which you want
+	// Amazon Redshift to provision the cluster.
+	// Default: A random, system-chosen Availability Zone in the region that is
+	// specified by the endpoint.
+	// Example: us-east-2d
+	// Constraint: The specified Availability Zone must be in the same region as
+	// the current endpoint. The Availability Zone parameter can't be specified
+	// if the MultiAZ parameter is set to true.
+	// The specified Availability Zone must be in the same AWS Region as the current endpoint.
+	// +optional
+	AvailabilityZone *string `json:"availabilityZone,omitempty"`
+
+	// ClusterParameterGroupName is the name of the DB cluster parameter group to use for the DB cluster.
+	// Default: The default Amazon Redshift cluster parameter group. For information
+	// about the default parameter group, go to Working with Amazon Redshift Parameter
+	// Groups (https://docs.aws.amazon.com/redshift/latest/mgmt/working-with-parameter-groups.html)
+	// +optional
+	ClusterParameterGroupName *string `json:"clusterParameterGroupName,omitempty"`
```

Comment:
Would it make sense to have  ParameterGroup as a separate custom resource which can then by attached to clusters?

Reply:
Yes certainly. But given the time and sheer number parameters it can accept, I decided to go ahead with this barebones structure for the `v1alpha1`. Leaving a little room for improvement around there for the next version iteration.